clusterloader2/api/validation.go

```diff
@@ -18,24 +18,186 @@ package api
 
 import (
 	"fmt"
+	"os"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/perf-tests/clusterloader2/pkg/errors"
 )
 
+// ConfigValidator contains metadata for config validation
+type ConfigValidator struct {
+	basePath   string
+	testConfig *Config
+}
+
+// NewConfigValidator creates a new ConfigValidator object
+func NewConfigValidator(basePath string, testConfig *Config) *ConfigValidator {
+	return &ConfigValidator{
+		basePath:   basePath,
+		testConfig: testConfig,
+	}
+}
+
 // Validate checks and verifies the configuration parameters.
-func (conf *Config) Validate() *errors.ErrorList {
-	return conf.Namespace.Validate()
+func (v *ConfigValidator) Validate() *errors.ErrorList {
+	allErrs := field.ErrorList{}
+	c := v.testConfig
+
+	// TODO: remove after deprecated automanagedNamespaces is disabled.
```

Comment:
I'm against creating ownerless/issue-less TODOs. Could you create an issue for this and link it to this TODO. THe issue can be super simply, just the title saying: "Remove deprecated automanagedNamespaces" or something like this

Reply:
Sure, shall I change this everywhere else in the codebase?

Reply:
Would be great, I hope there is just a few of such comments, right?

Reply:
Not too many :)